External/Plugins/ASCompletion/Completion/ASComplete.cs

```diff
@@ -1782,66 +1782,38 @@ private static ClassModel ResolveParameterType(int paramIndex, bool indexTypeOnl
         /// <summary>
         /// Locate beginning of function call parameters and return index of current parameter
         /// </summary>
-        internal static int FindParameterIndex(ScintillaControl Sci, ref int position)
+        internal static int FindParameterIndex(ScintillaControl sci, ref int position)
         {
             int parCount = 0;
             int braCount = 0;
             int comaCount = 0;
             int arrCount = 0;
-            var genCount = 0;
-            var dquCount = 0;
-            var squCount = 0;
+            var genCount = 0;
+            sci.Colourise(0, -1);
```

Comment:
My understanding is that calling `Colourise(0, -1)` is expensive, and some of the functions calling `FindParameterIndex` are already calling this before (although it's not easy to find some of them by just browsing the code). @elsassph @Meychi is my comment right or am I remembering wrong/something else? If so then I guess it would be better to identify which functions calling this are not calling `Colourise` before?

Other than this I think it's ok.

Reply:
We could add some logging of calls to `Colourise` to check if we're abusing it yes.